sotodlib/core/metadata/loader.py

```diff
@@ -452,22 +491,42 @@ def check_tags(det_info, aug_request, final=False):
 
             # Make everything an axisman.
             if isinstance(item, ResultSet):
+                # Note this might raise an IncompleteDetInfoError.
                 item = broadcast_resultset(item, det_info=det_info)
 
             elif not isinstance(item, core.AxisManager):
                 logger.error(
                     f'The decoded item {item} is not an AxisManager or '
                     f'other well-understood type.  Request was: {request}.')
 
+            # You have to check for detector loss here -- compare
+            # item.dets.vals to what's in det_info.
+            i0 = core.util.get_multi_index(
+                item.dets.vals, det_info['readout_id'])
+
+            n_dets_item = len(set(i0[i0>=0]))
+            if n_dets_item < n_dets:
+                message = (f"Only {n_dets_item} of {n_dets} detectors "
+                           "have data for metadata specified by "
+                           f"spec={_spec}. ")
+                if _on_missing == 'trim':
+                    logger.warning(message + 'Trimming.')
+                elif _on_missing == 'fail':
+                    raise IncompleteMetadataError(message)
+                else:  # skip
+                    logger.warning(message + 'Discarding.')
+                    continue
+
             # Unpack it.
             try:
-                unpackers = Unpacker.decode(spec['name'], item)
-                for unpacker in unpackers:
-                    dest = unpacker.unpack(item, dest=dest)
+                dest = unpack_item(spec.unpack, item, dest=dest)
+                #unpacker = _Unpacker.from_spec(spec, target=item)
```

Comment:
hanging commented lines?

Reply:
Sorry, forgot to push this cleanup commit -- will include in my next PR.